apps/ejabberd/src/ejabberd_auth_jwt.erl

```diff
@@ -57,7 +57,26 @@
 %%%----------------------------------------------------------------------
 
 -spec start(Host :: ejabberd:server()) -> ok.
-start(_Host) ->
+start(Host) ->
+    UsernameKey = get_auth_opt(Host, jwt_username_key),
+    true = is_atom(UsernameKey) andalso UsernameKey /= undefined,
+
+    JWTSecret =
+    case {get_auth_opt(Host, jwt_secret_source), get_auth_opt(Host, jwt_secret)} of
```

Comment:
I agree with @arcusfelis moving this case to a separate function will improve readability.